app/controllers/budgets_controller.rb

```diff
@@ -1,50 +1,56 @@
 class BudgetsController < ApplicationController
-  before_action :authenticate_user!
   before_action :set_budget, only: %i[show edit update destroy]
+  before_action :set_categories_and_months, only: %i[new create edit update]
 
   def index
     @budgets = current_user.budgets
+    @budgets = @budgets.by_month(params[:month]) if params[:month].present?
+    @budgets = @budgets.by_category(params[:category_id]) if params[:category_id].present?
   end
 
-  def show
-  end
+  def show; end
 
   def new
-    @budget = current_user.budgets.build
+    @budget = Budget.new
   end
 
   def create
     @budget = current_user.budgets.build(budget_params)
     if @budget.save
-      redirect_to root_path, notice: 'Budget was successfully created.'
+      redirect_to budgets_path, notice: 'Budget was successfully created.'
     else
       render :new
     end
   end
 
-  def edit
-  end
+  def edit; end
 
   def update
     if @budget.update(budget_params)
-      redirect_to @budget, notice: 'Budget was successfully updated.'
+      redirect_to budgets_path, notice: 'Budget was successfully updated.'
     else
       render :edit
     end
   end
 
   def destroy
     @budget.destroy
-    redirect_to budgets_url, notice: 'Budget was successfully destroyed.'
+    redirect_to budgets_path, notice: 'Budget was successfully destroyed.'
   end
 
   private
 
   def set_budget
     @budget = current_user.budgets.find(params[:id])
+  rescue ActiveRecord::RecordNotFound
+    budget_not_found
```

Comment:
You are rescuing from this exception in application controller. No need to have it here as well.